bundles/org.openhab.core.model.yaml/src/main/java/org/openhab/core/model/yaml/internal/semantics/YamlSemanticTags.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.model.yaml.internal.semantics;
+
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.core.model.yaml.YamlElement;
+import org.openhab.core.model.yaml.YamlFile;
+import org.openhab.core.model.yaml.YamlParseException;
+
+/**
+ * The {@link YamlSemanticTags} is a data transfer object used to serialize a list of semantic tags
+ * in a YAML configuration file.
+ *
+ * @author Laurent Garnier - Initial contribution
+ */
+@NonNullByDefault
+public class YamlSemanticTags extends YamlFile {
+
+    public List<YamlSemanticTag> tags = List.of();
+
+    @Override
+    public List<? extends YamlElement> getElements() {
+        return tags;
+    }
+
+    @Override
+    protected void checkVersion() throws YamlParseException {
```

Comment:
I think the version check in this way will not work. If the format changes, the YAML reader will fail before you get to parse the version. I think you would need to use the generic reader, extract the version and check it in the model repository. It would probably be better to just return the supported version here (`int getVersion()`). You could leave that a dummy for now.

Reply:
I removed `checkVersion` and added getter `getVersion`.